<launch>

  <arg name="robot_name" default="my_gen3" />
  <!-- Launch RealSense camera -->
  <include file="$(find realsense2_camera)/launch/rs_camera.launch" />

  <group ns="my_gen3_lite">
    <param name="robot_description"
      command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/gen3_lite_gen3_lite_2f.xacro sim:=false" />
    <!-- The semantic description that corresponds to the URDF -->
    <include file="$(find gen3_lite_gen3_lite_2f_move_it_config )/launch/move_group.launch" />
  </group>

  <!-- Launch ArUco detection node -->
  <node name="aruco_detect" pkg="aruco_detect" type="aruco_detect" output="screen">
    <remap from="image" to="/camera/color/image_raw" />
    <param name="dictionary" value="DICT_6X6_250" />
  </node>

  <!-- Launch custom capture and process image node -->
  <node name="capture_and_process_image_node" pkg="kinova_5551_chess_arm"
    type="capture_and_process_image.py" />

  <!-- Launch KinovaChessControl node -->
  <node name="kinova_chess_control" pkg="kinova_5551_chess_arm" type="KinovaChessControl.py"
    output="screen" />
</launch>